### PR TITLE
feat: add optional explanation field to be displayed on clicking show answer

### PR DESCRIPTION
**Pull request by kaustavb12**
## Description

This PR:

1. adds an optional `Explanation` text area field in the edit popup of studio. When a text is added here, its displayed to the learner below the xblock when `Show Answer` button is pressed.
image

2. fixes an issue where the feedback div was displayed even when no feedback content was present. This resulted an extra line being displayed because of the top margin of the feedback div.

Before:
image

After:
image

## Supporting information

OpenCraft internal ticket : [BB-6464](https://tasks.opencraft.com/browse/BB-6464)

## Testing instructions

1. Checkout this branch in <devstack_parent>/src directory in your local setup
2. Login to Studio shell, switch to edxapp virtual env
4. Go to /edx/src/xblock-drag-and-drop-v2 and run pip install -e .
5. Go to Studio UI and create a new unit
6. Select Problem > Advanced > Drag and Drop, to create a v2 drag and drop xblock
7. Click Edit and change mode to Assessment
8. Scroll down to `Explanation Text`
9. Add some text in the text area
10. Click Continue twice and Save
11. Click on `View Live Version`
12. Ensure there is no extra line as show above
13. Do as many attempts as required till `Show Answer` button is active
14. Click on `Show Answer`
15. Ensure `Explanation` is displayed below the xblock

## Deadline

"None"

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @kaustavb12!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @kaustavb12!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by natabene** [comment]
@kaustavb12 Thank you for your contribution. @Agrendalath Thanks for checking, we are fine, please review and merge as you find fit.

File: drag_and_drop_v2/drag_and_drop_v2.py [hunk 1]
```diff
@@ -618,7 +618,12 @@ def show_answer(self, data, suffix=''):
                 self.i18n_service.gettext("There are attempts remaining")
             )
 
-        return self._get_correct_state()
+        answer = self._get_correct_state()
+
+        if self.data.get('explanation'):
+            answer['explanation'] = self.data.get('explanation')
```
[the remark this answers is not in the document]
Comment (kaustavb12) [reply drag_and_drop_v2/drag_and_drop_v2.py]: Done

**Comment by kaustavb12** [comment]
@Agrendalath I have added the tests and the translations

**Review by Agrendalath — APPROVED** [review]
:+1:

* [x] I tested this: verified that the explanations are displayed correctly
* [x] I read through the code
* [x] I checked for accessibility issues
* [x] Includes documentation: n/a
* [x] I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository: n/a

**Comment by openedx-webhooks** [comment]
@kaustavb12 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/xblock-drag-and-drop-v2&entry.867055334=https://github.com/openedx/xblock-drag-and-drop-v2/pull/289&entry.1484655318=https://github.com/kaustavb12&entry.752974735=2022-07-17+04:16&entry.1917517419=2022-07-22+00:36&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by openedx-webhooks** [comment]
@kaustavb12 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/xblock-drag-and-drop-v2&entry.867055334=https://github.com/openedx/xblock-drag-and-drop-v2/pull/289&entry.1484655318=https://github.com/kaustavb12&entry.752974735=2022-07-17+04:16&entry.1917517419=2022-07-22+00:36&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**